app/models/ability.rb

```diff
@@ -0,0 +1,48 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # return unless user.present?
+    # can :read, :all
+
+    # can %i[create update destroy], Post, author_id: user.id
+    # can %i[create update destroy], Comment, author_id: user.id
+
+    # return unless user.is? :admin
+    # can :manage, :all
+
+    user ||= User.new # guest user (not logged in)
+    if user.is? :admin
+      can :manage, :all
+    else
+      can :read, :all
+      can :create, Post, author_id: user.id
+      can :create, Comment, :all
+    end
```

Comment:
- [ ] Please good Job by mentioning the post and comment in the ability but as it was requested by the project requirements, please the user  can manage all the posts and comments if they are his own comments and posts so consider changing create to manage currently the user can just create them